apps/admin/backend/src/server.ts

```diff
@@ -109,528 +161,264 @@ function buildApi(auth: DippedSmartCardAuthApi, workspace: Workspace) {
-    async (request, response) => {
-      /* istanbul ignore next */
-      const file = !Array.isArray(request.files)
-        ? request.files?.[CVR_FILE_ATTACHMENT_NAME]?.[0]
-        : undefined;
-      try {
-        const { electionId } = request.params;
-
-        if (!file) {
-          response.status(400).json({
-            status: 'error',
-            errors: [
-              {
-                type: 'invalid-value',
-                message: `expected file field to be named "${CVR_FILE_ATTACHMENT_NAME}"`,
-              },
-            ],
-          });
-          return;
+    async configure(input: { electionData: string }): Promise<ConfigureResult> {
+      const parseResult = safeParseElectionDefinition(input.electionData);
+      if (parseResult.isErr()) {
+        return err({ type: 'parsing', message: parseResult.err().message });
+      }
+      const electionDefinition = parseResult.ok();
+      const electionId = store.addElection(electionDefinition.electionData);
+      store.setCurrentElectionId(electionId);
+      await logger.log(
+        LogEventId.ElectionConfigured,
+        assertDefined(await getUserRole()),
+        {
+          disposition: 'success',
+          newElectionHash: electionDefinition.electionHash,
         }
+      );
+      return ok({ electionId });
+    },
 
-        const parseQueryResult = safeParse(
-          Admin.PostCvrFileQueryParamsSchema,
-          request.query
-        );
-
-        if (parseQueryResult.isErr()) {
-          response.status(400).json({
-            status: 'error',
-            errors: [
-              {
-                type: 'ValidationError',
-                message: parseQueryResult.err().message,
-              },
-            ],
-          });
-          return;
+    async unconfigure(): Promise<void> {
+      store.deleteElection(loadCurrentElectionIdOrThrow(workspace));
+      store.setCurrentElectionId();
+      await logger.log(
+        LogEventId.ElectionUnconfigured,
+        assertDefined(await getUserRole()),
+        {
+          disposition: 'success',
         }
+      );
+    },
 
-        const { analyzeOnly } = parseQueryResult.ok();
+    getCurrentElectionMetadata(): Admin.ElectionRecord | null {
+      const currentElectionId = store.getCurrentElectionId();
+      if (currentElectionId) {
+        const electionRecord = store.getElection(currentElectionId);
+        assert(electionRecord);
+        return electionRecord;
+      }
 
-        const parseBodyResult = safeParse(
-          Admin.PostCvrFileRequestSchema,
-          request.body
-        );
-        if (parseBodyResult.isErr()) {
-          response.status(400).json({
-            status: 'error',
-            errors: [
-              {
-                type: 'invalid-request-body',
-                message: parseBodyResult.err().message,
-              },
-            ],
-          });
-          return;
-        }
+      return null;
```

Comment:
I remember changing it to `null` because I thought `grout` was complaining about `undefined`. Maybe that was unrelated, I'll see if I can change it back.

Reply:
ok yeah it can just be `undefined`, not sure what I hit before. switched.

Reply:
Now I'm reminded why I switched it, I get this error:

```Query data cannot be undefined. Please make sure to return a value other than undefined from your query function. Affected query key: ["getCurrentElectionMetadata"]```

Seems like a React Query requirement, presumable because the data being `undefined` can indicate something else (no data).

Reply:
I see there are tests in `grout.test.ts` for endpoints that return `Person | undefined` but maybe that's not actually supported?

Reply:
Oh yeah I do recall learning that limitation of react-query at some point. It's definitely a react-query limitation, not a grout limitation - grout supports `undefined`. Maybe just add a comment explaining why?